app/views/dashboard/autoyast.xml.erb

```diff
@@ -9,7 +9,12 @@
         <interpreter>shell</interpreter>
         <source><![CDATA[
         #!/bin/sh
-        echo "master: <%= @controller_node %>" > /etc/salt/minion.d/master.conf
+        MINION_CFG="/etc/salt/minion.d/master.conf"
+        [ -f $MINION_CFG ] && MINION_ID=$(grep id $MINION_CFG | sed 's/id: //')
```

Comment:
Ok, as you want. I am fan of validating everything but you are right that this will be run once only during installation.